modules/002-deckhouse/hooks/telemetry_update_window_test.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2022 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/pointer"
+
+	"github.com/deckhouse/deckhouse/go_lib/telemetry"
+	. "github.com/deckhouse/deckhouse/testing/hooks"
+)
+
+var _ = Describe("Modules :: deckhouse :: hooks :: telemetry :: update window", func() {
+	f := HookExecutionConfigInit(`{
+        "global": {
+          "modulesImages": {
+			"registry": "my.registry.com/deckhouse"
+		  }
+        },
+		"deckhouse": {
+              "internal": {},
+              "releaseChannel": "Stable",
+			  "update": {
+				"mode": "Auto",
+				"windows": [{"from": "00:00", "to": "23:00"}]
+			  }
+			}
+}`, `{}`)
+
+	assertModeMetric := func(f *HookExecutionConfig, typeT string) {
+		metrics := f.MetricsCollector.CollectedMetrics()
+
+		metricIndex := -1
+		for i, m := range metrics {
+			if m.Name == telemetry.WrapName("update_window_approval_mode") {
+				Expect(m.Value).To(Equal(pointer.Float64Ptr(1.0)))
+				Expect(m.Labels).To(HaveKey("mode"))
+				Expect(m.Labels["mode"]).To(Equal(typeT))
+				metricIndex = i
+				break
+			}
+		}
+
+		Expect(metricIndex >= 0).To(BeTrue())
+	}
+
+	assertWindowMetric := func(f *HookExecutionConfig, from, to, day, human string) {
+		metrics := f.MetricsCollector.CollectedMetrics()
+
+		Expect(metrics).ToNot(BeEmpty())
+
+		metricIndex := -1
+		dayFound := false
+		for i, m := range metrics {
+			if m.Name == telemetry.WrapName("update_window") {
```

Comment:
In tests, we should have the full explicit name of the metric. If somebody changes the name wrapper, we might not notice the issue. I suggest to avoid calling telemetry.WrapName in tests

Reply:
done